static/js/giphy.js

```diff
@@ -43,8 +52,23 @@ function renderGIPHYGrid(targetEl) {
                 // GIF; nice in principle but too distracting.
                 hideAttribution: true,
                 onGifClick: (props) => {
-                    $("#compose_box_giphy_grid").popover("hide");
-                    compose_ui.insert_syntax_and_focus(`[](${props.images.downsized_medium.url})`);
+                    hide_giphy_popover();
+                    if (edit_message_id !== undefined) {
+                        const edit_message_textarea = $(
+                            `#message_edit_content_${CSS.escape(edit_message_id)}`,
+                        );
+                        // Assign null to edit_message_id so that the selection of GIFs in new
+                        // message composition form works correctly.
+                        edit_message_id = undefined;
+                        compose_ui.insert_syntax_and_focus(
+                            `[](${props.images.downsized_medium.url})`,
+                            edit_message_textarea,
+                        );
+                    } else {
+                        compose_ui.insert_syntax_and_focus(
+                            `[](${props.images.downsized_medium.url})`,
+                        );
+                    }
```

Comment:
We should simplify this to to not call `compose_ui.insert_syntax_and_focus` in both code paths, and instead just have the conditional determine what what a `textarea` parameter we pass should be.

Reply:
Done